src/Compilers/CSharp/Portable/Binder/Semantics/OverloadResolution/OverloadResolutionResult.cs

```diff
@@ -44,10 +44,7 @@ public bool Succeeded
         {
             get
             {
-                if (!_bestResultState.HasValue())
-                {
-                    _bestResultState = TryGetBestResult(this.ResultsBuilder, out _bestResult);
-                }
+                EnsureBestResultLoaded();
```

Comment:
What is the perf impact of this method?  It seems like we'd only delay load this value if it was expensive and now we're promptly loading it in a lot more places.